lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -453,30 +462,62 @@ async function validateQuotas(bucket, account, apiNames, apiMethod, inflight, lo
             }
         }
 
+        if (actionWithDataDeletion[apiMethod]) {
+            const responseTimeInNs = Number(process.hrtime.bigint() - requestStartTime);
+            monitoring.quotaEvaluationDuration.labels({
+                action: apiMethod,
+                type: 'delete',
+                code: 200,
+            }).observe(responseTimeInNs / 1e9);
+            return callback();
+        }
+
         if (bucketQuotaExceeded || accountQuotaExceeded) {
-            if (apiMethod?.endsWith('Delete')) {
-                return callback();
-            }
             // clean any inflight bytes
             if (inflight > 0) {
                 // eslint-disable-next-line no-await-in-loop
-                await ScubaClientInstance.getLatestMetrics('bucket',
+                await ScubaClientInstance.getUtilizationMetrics('bucket',
                     `${bucket.getName()}_${creationDate}`, null, {
                     action: apiMethod,
                     inflight: -inflight,
                 });
                 if (account?.quota) {
                     // eslint-disable-next-line no-await-in-loop
-                    await ScubaClientInstance.getLatestMetrics('account',
+                    await ScubaClientInstance.getUtilizationMetrics('account',
                         `${account.account}_${creationDate}`, null, {
                         action: apiMethod,
                         inflight: -inflight,
                     });
                 }
             }
+            const responseTimeInNs = Number(process.hrtime.bigint() - requestStartTime);
+            let type = 'bucket';
+            if (accountQuotaExceeded && bucketQuotaExceeded) {
+                type = 'bucketAccount';
+            } else if (accountQuotaExceeded) {
+                type = 'account';
+            }
+            monitoring.quotaEvaluationDuration.labels({
+                action: apiMethod,
+                type,
+                code: 429,
+            }).observe(responseTimeInNs / 1e9);
             return callback(errors.QuotaExceeded);
         }
 
+        const responseTimeInNs = Number(process.hrtime.bigint() - requestStartTime);
+        let type = 'bucket';
+        if (bucketQuota && accountQuota) {
+            type = 'bucketAccount';
+        } else if (accountQuota) {
+            type = 'account';
+        }
+        monitoring.quotaEvaluationDuration.labels({
+            action: apiMethod,
+            type,
+            code: 200,
+        }).observe(responseTimeInNs / 1e9);
+
         return callback();
     } catch (err) {
```

Comment:
Should you observe the quota evaluation duration in the catch for errors from Scuba ? I don't see how you got some Bad Requests in your graph otherwise

Reply:
Done in https://github.com/scality/cloudserver/pull/5556/commits/0006a0f07d83cb818352e0f6caec6662323aee19

I chose the error code 202, because in case of error we still process the request. Note that the error from scubaclient is also monitored, so information is not lost.